Documented queue storage in handbook (#753)
Altinn/altinn-studio-docs: content/community/contributing/handbook/azure-storage/queues/_index.md

---
title: Queues
description: Guidelines for working with Azure Queue Storage
tags: [development, operations, storage, azure, queue]
weight: 100,
toc: true
---

## Moving messages between queues

During operations we sometimes need to move messages between queues. 

> The most common scenario is wanting to re-process elements that have ended in the poision 

Reviewer: ```suggestion
The most common scenario is wanting to re-process elements that have ended in the poison 
```

queue after too many failed attempts. 

### Prerequisites
- [Azure Storage Explorer](../tools/#azure-storage-explorer) 
- An Azure AD account with write-permissions for the related storage account 

### How to
1. Open Storage Explorer.
    If you are working towards a test or production environment you must 
      - Sign into Azure under the Account Management tab
      - select the storage account you want to work with 
2. Move to the _Explorer_ tab and identify the source queue in the left hand menu
    ![Poison queue marked in red](identify-poisonqueue.png)
3. You should now see the messages in the queue in the explorer.

>     If you only want to move a single message, this should be highlightes

Reviewer: ```suggestion
    If you only want to move a single message, this should be highlighted
```

    ![Poision queue with elements](move-message-btn-marked.png)
4. Click the arrow next to _Move messages_ to specify if all or just a single message should be moved
5. In the dialogue, select the destination queue and click `Move`
    ![Select destination queue dialogue](select-destination-queue.png)

> 6. The messages will be moved immediatly, check the destination queue to confirm.

Reviewer: ```suggestion
6. The messages will be moved immediatly. Check the destination queue to confirm.
```

## Manually adding messages in queue

> During development it can be useful to manually add messages to a queue and during incident management it might also 
> be required to do this during operations as well. 

Reviewer: ```suggestion
Manually adding a message to a queue is useful both when testing new functionality and during incident management.
```
